### PR TITLE
fix: collect to unlock changes

### PR DESCRIPTION
**Pull request by intergalacticspacehighway**
# Why
https://linear.app/showtime/issue/SHOW2-2007/collect-to-unlock-channel-small-product-changes

# How
Gets NFT details, edition by slug, makes sure that performance is only affected for users who are not subscribed in the channel. We won't need API call once we receive edition in the API. Reported to backend!

# Test Plan
- Try collecting to unlock button in channel

### CODE REVIEW

File: packages/app/hooks/use-infinite-list-query.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export const useInfiniteListQuerySWR = <T>(
       // suspense: true,
       refreshInterval,
       revalidateOnMount: true,
-      dedupingInterval: 30000,
+      dedupingInterval: 5000,
```
Comment (intergalacticspacehighway) [inline packages/app/hooks/use-infinite-list-query.ts:45]: this was required or else doing mutate doesn't perform refetch (we need to refetch once user has collected the star drop from default selected card). I think 5 seconds is good enough for dedup interval. cc - @hirbod let me know what you think!

Comment (hirbod) [reply packages/app/hooks/use-infinite-list-query.ts:45]: Thats fine. I didn't know it had that many implications

File: packages/app/components/creator-channels/messages.tsx [hunk 2]
```diff
@@ -536,7 +545,7 @@ export const Messages = memo(() => {
           editMessage={editMessage}
           isUserAdmin={isUserAdmin}
           keyboard={keyboard}
-          latestPaidNFTSlug={latest_paid_nft_slug}
+          edition={edition}
```
Comment (hirbod) [inline packages/app/components/creator-channels/messages.tsx:548]: why did you remove latestPaidNFTSlug? This is required for the different states inside the MessageInput component

Comment (hirbod) [reply packages/app/components/creator-channels/messages.tsx:548]: Ok I have seen the changes, but is that safe enough?

Comment (intergalacticspacehighway) [reply packages/app/components/creator-channels/messages.tsx:548]: yeah should be good. Not perfect. Ideal solution would be to wait for BE to add edition in the collection detail API. We can re-iterate once they do. wdyt?

File: packages/app/hooks/use-infinite-list-query.ts (hunk 1 above)
Comment (intergalacticspacehighway) [reply packages/app/hooks/use-infinite-list-query.ts:45]: yeah, it is not obvious. So if mutate is triggered within 30 secs SWR counts it as a duplicate request and doesn't refetch. Took a while to debug

File: packages/app/components/creator-channels/messages.tsx (hunk 2 above)
Comment (hirbod) [reply packages/app/components/creator-channels/messages.tsx:548]: But I am using the channel by detail endpoint to determine if there is any paid nft, why should there be the need to use the edition for it?

Comment (intergalacticspacehighway) [reply packages/app/components/creator-channels/messages.tsx:548]: We need the edition to trigger the pay flow directly instead of going to the drop page on `Collect to unlock` button tap. The edition is required by our Pay button component. If the channel by detail endpoint can also return star drop edition that would save us 2 requests that I am [making.](https://github.com/showtime-xyz/showtime-frontend/pull/2345/files#diff-35e8b62f9b70f65e462c40a883cc47ea668799d3de4585b0d46b0c098b8b6b8fR189)

**Review by alantoa — APPROVED** [review]
LGTM 🔥
